src/components/navigation/SlideMenuContext.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react';
```

Comment:
That space is more for contexts that should be used in multiple places throughout the application. This context should only ever be used by the slide panel to keep centralized state so I didn't include it in there

Reply:
Gotcha, in that case you're right to keep it in this directory.

Just curious, but if the logic here is only used by a single component and it's direct children (SlideMenu and SlideMenuItem) why use a context instead of keeping state in the SlideMenu component? any future plans for this context?